pkg/apis/contrail/v1alpha1/base_types.go

```diff
@@ -1015,35 +968,63 @@ func (c *ControlClusterConfiguration) FillWithDefaultValues() {
 
 // ZookeeperClusterConfiguration defines all configuration knobs used to write the config file.
 type ZookeeperClusterConfiguration struct {
-	ClientPort               string
-	ServerPort               string
-	ElectionPort             string
-	ServerListCommaSeparated string
-	ServerListSpaceSeparated string
+	ClientPort   int      `json:"clientPort,omitempty"`
+	ServerPort   int      `json:"serverPort,omitempty"`
+	ElectionPort int      `json:"electionPort,omitempty"`
+	ServerIPList []string `json:"serverIPList,omitempty"`
+}
+
+// FillWithDefaultValues fills Zookeeper config with default values
+func (c *ZookeeperClusterConfiguration) FillWithDefaultValues() {
+	if c.ClientPort == 0 {
+		c.ClientPort = ZookeeperPort
+	}
+	if c.ElectionPort == 0 {
+		c.ElectionPort = ZookeeperElectionPort
+	}
+	if c.ServerPort == 0 {
+		c.ServerPort = ZookeeperServerPort
+	}
 }
 
 // RabbitmqClusterConfiguration defines all configuration knobs used to write the config file.
 type RabbitmqClusterConfiguration struct {
-	Port                                string
-	SSLPort                             string
-	ServerListCommaSeparated            string
-	ServerListSpaceSeparated            string
-	ServerListCommaSeparatedSSL         string
-	ServerListSpaceSeparatedSSL         string
-	ServerListCommaSeparatedWithoutPort string
-	Secret                              string
+	Port       int      `json:"port,omitempty"`
+	SSLPort    int      `json:"sslPort,omitempty"`
+	ServerList []string `json:"serverList"`
```

Comment:
ServerIPList

Reply:
why is there no omitempty here and it is in other structs? If it is better than remove it in others as well.

Reply:
Done